docs: rework README, add page about ignoring diagnostics (#453)
denoland/deno_lint: www/public/ignoring-rules.md

## Ignore directives

### Files

> To ignore whole file use `// deno-lint-ignore-file` directive at the top of the file.

Reviewer: ```suggestion
To ignore a whole file use `// deno-lint-ignore-file` at the top of the file:
```

```ts
// deno-lint-ignore-file

function foo(): any {
  // ...
}
```

> Ignore directive must be placed before first stament or declaration:

Reviewer: ```suggestion
The ignore directive must be placed before the first statement or declaration:
```

Author: Thanks @RDambrosio016, I applied your suggestions

```ts
// Copyright 2020 the Deno authors. All rights reserved. MIT license.

/**
 * Some JS doc
 **/

// deno-lint-ignore-file

import { bar } from "./bar.js";

function foo(): any {
  // ...
}
```

> You can also ignore certain diagnostics in the whole file

Reviewer: ```suggestion
You can also ignore certain diagnostics in the whole file:
```

```ts
// deno-lint-ignore-file no-explicit-any no-empty

function foo(): any {
  // ...
}
```

### Diagnostics

> To ignore specific diagnostic use `// deno-lint-ignore <codes...>` directive on the preceding

Reviewer: ```suggestion
To ignore specific diagnostics use `// deno-lint-ignore <codes...>` on the preceding
```

line of the offending line.

```ts
// deno-lint-ignore no-explicit-any
function foo(): any {
  // ...
}

// deno-lint-ignore no-explicit-any explicit-function-return-type
function bar(a: any) {
  // ...
}
```

> Specyfing rule name that should be ignored is required.

Reviewer: ```suggestion
You must specify the names of the rules to be ignored.
```
